rustworkx-core/src/connectivity/find_cycle.rs

```diff
@@ -54,25 +56,30 @@ use std::hash::Hash;
 /// ```
 pub fn find_cycle<G>(graph: G, source: Option<G::NodeId>) -> Vec<(G::NodeId, G::NodeId)>
 where
+    G: Copy,
```

Comment:
This seems like a pretty big API compatibility issue. I wouldn't expect most graph types to by `Copy`. `Clone` sure, but `Copy` seems to go against the spirit of the trait to use if for graph objects.

Reply:
I fixed it, with `kosaraju_scc` and references I don't need copy anymore